hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/MapValuesWithPagingPredicateMessageTask.java

```diff
@@ -39,7 +41,14 @@ public MapValuesWithPagingPredicateMessageTask(ClientMessage clientMessage, Node
 
     @Override
     protected Object reduce(Collection<QueryResultRow> result) {
-        return getSortedPageEntries(result, parameters.predicate);
+        Map.Entry<PagingPredicateImpl, List<Map.Entry<Data, Data>>> sortResult = getSortedPageEntries(result,
+                parameters.predicate);
+
+        List<Map.Entry<Data, Data>> entries = sortResult.getValue();
+        List<Data> dataList = new ArrayList<>(entries.size());
+        entries.forEach(entry -> dataList.add(entry.getValue()));
```

Comment:
Again, it would be to get this list directly in `getSortedPageEntries()` and not create a second list.

Reply:
done.

Reply:
I see we still create a separate `valueList` list from the result of `getSortedPageEntries()` but let's leave it as it is. This can be optimized later if it proves to be a bottleneck.